### DISCUSSION TITLE
Sending an Encoded Base64 Image to  LLaMa 3.2 Vision Instruct 90B

### DISCUSSION THREAD

**Discussion by adamocarolli** [sgl-project/sglang#2227] [Q&A] [answered]
Hi found this for sending an image url which worked: https://sgl-project.github.io/backend/openai_api_vision.html#Using-OpenAI-Python-Client

However, I'm unable to encode an image and send it to the processor it seems to get terminated via:

```
File "/sgl-workspace/sglang/python/sglang/srt/managers/image_processor.py", line 228, in process_images_async
    image_inputs = await self._process_single_image(images, input_text)
  File "/sgl-workspace/sglang/python/sglang/srt/managers/image_processor.py", line 199, in _process_single_image
    image_inputs = await loop.run_in_executor(
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
[2024-11-27 12:08:03] INFO:     Waiting for application shutdown.
```

Any documentation or examples on encoding the image properly?

**Comment by gxm651182644** [comment]
i have the same issue when i query qwen2-vl-2b-instruct model

**Comment by gxm651182644** [comment]
> Hi found this for sending an image url which worked: https://sgl-project.github.io/backend/openai_api_vision.html#Using-OpenAI-Python-Client
> 
> However, I'm unable to encode an image and send it to the processor it seems to get terminated via:
> 
> ```
> File "/sgl-workspace/sglang/python/sglang/srt/managers/image_processor.py", line 228, in process_images_async
>     image_inputs = await self._process_single_image(images, input_text)
>   File "/sgl-workspace/sglang/python/sglang/srt/managers/image_processor.py", line 199, in _process_single_image
>     image_inputs = await loop.run_in_executor(
> concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
> [2024-11-27 12:08:03] INFO:     Waiting for application shutdown.
> ```
> 
> Any documentation or examples on encoding the image properly?

do you solve the problem?

**Reply by adamocarolli** [reply]
Yeah this is the problem I'm running into as well; I figured this may be because my cluster does not have access to internet. Does yours have access to internet?

**Comment by muziyongshixin** [comment, marked as answer]
you can just send the image content via base64 encoding，like the code bellow:

```
import base64

def image_to_data_url(img_path):
    with open(img_path, "rb") as f:
        data = base64.b64encode(f.read()).decode("utf-8")
    # 注意 MIME 类型要改成实际图片类型（jpeg/png/webp等）
    return f"data:image/jpeg;base64,{data}"

messages = [
    {
        "role": "user",
        "content": [
            {"type": "text", "text": "这是一张测试图片"},
            {
                "type": "image_url",
                "image_url": {
                    "url": image_to_data_url("/path/to/your/image.jpg")
                }
            }
        ]
    }
]
```
